synthesizer/benches/utilities/workload.rs

```diff
@@ -0,0 +1,182 @@
+        let mut root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        root.push(".resources");
+        root.push(&name);
+        // Construct the object store.
+        let object_store = ObjectStore::new(root)?;
+        // Construct the workload.
+        Ok(Self { name, benchmarks, object_store })
+    }
+
+    /// Adds a benchmark to the workload.
+    pub fn add(&mut self, benchmark: Box<dyn Benchmark<Testnet3>>) {
+        self.benchmarks.push(benchmark);
+    }
+
+    /// Returns the name of the workload.
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+
+    /// Constructs batches of setup transactions and benchmark transactions from the benchmarks in the workload.
+    /// Note that setup operations are aggregated across all benchmarks.
+    pub fn setup<C: ConsensusStorage<Testnet3>>(
+        &mut self,
+    ) -> (VM<Testnet3, C>, PrivateKey<Testnet3>, BenchmarkBatch, TestRng) {
+        println!("Setting up workload: {}", self.name);
+        // Check that the seed to the RNG is stored in the object store.
+        let mut all_data_is_stored = self.object_store.contains("seed");
+        // Check that the relevant blocks are stored in the object store.
+        all_data_is_stored &= match self.object_store.get("num_blocks") {
+            Err(err) => false,
+            Ok(num_blocks) => {
+                let num_blocks: u64 = num_blocks;
+                (0..num_blocks).all(|i| self.object_store.contains(format!("block_{}", i)))
+            }
+        };
+        // Check that the benchmark transactions are stored in the object store.
+        for benchmark in &self.benchmarks {
+            all_data_is_stored &= self.object_store.contains(benchmark.name())
+        }
+
+        // If all of the required items are not stored, clear the object store and initialize them.
+        if !all_data_is_stored {
+            println!("Data is not stored. Initializing...");
+            let mut setup_operations = vec![];
+            let mut benchmark_operations = vec![];
+
+            // Collect the operations for each benchmark.
+            for benchmark in &mut self.benchmarks {
+                let setup_ops = benchmark.setup_operations();
+                let benchmark_ops = benchmark.benchmark_operations();
+                setup_operations.push(setup_ops);
+                benchmark_operations.push((benchmark.name(), benchmark_ops));
+            }
+
+            // Fold the batches of setup operations across all benchmarks into a single sequence of batches.
+            let max_num_batches = setup_operations.iter().map(|operations| operations.len()).max().unwrap_or(0);
+            let mut aggregated_setup_operations = iter::repeat_with(Vec::new).take(max_num_batches).collect_vec();
+            for setup_ops in setup_operations {
+                for (i, operations) in setup_ops.into_iter().enumerate() {
+                    aggregated_setup_operations[i].extend(operations);
+                }
+            }
```

Comment:
am I misreading this, or is `aggregated_setup_operations` the same as `setup_operations`? even if it's a different collection, `let aggregated_setup_operations = setup_ops.into_iter().collect();` would suffice here, no?